src/merkle/simple_smt/mod.rs

```diff
@@ -244,23 +238,23 @@ impl Store {
         self.leaves
             .get(&key)
             .cloned()
-            .ok_or(MerkleError::InvalidIndex(self.depth, key))
+            .ok_or(MerkleError::InvalidIndex(NodeIndex::new(self.depth, key)))
     }
 
     fn insert_leaf_node(&mut self, key: u64, node: Word) {
         self.leaves.insert(key, node);
     }
 
-    fn get_branch_node(&self, key: u64, depth: u32) -> Result<BranchNode, MerkleError> {
+    fn get_branch_node(&self, node: &NodeIndex) -> Result<BranchNode, MerkleError> {
         self.branches
-            .get(&(key, depth))
+            .get(node)
             .cloned()
-            .ok_or(MerkleError::InvalidIndex(depth, key))
+            .ok_or(MerkleError::InvalidIndex(*node))
     }
 
-    fn insert_branch_node(&mut self, key: u64, depth: u32, left: RpoDigest, right: RpoDigest) {
-        let node = BranchNode { left, right };
-        self.branches.insert((key, depth), node);
+    fn insert_branch_node(&mut self, node: NodeIndex, left: RpoDigest, right: RpoDigest) {
```

Comment:
nit: I'd change `node` to `index` here as well.

Reply:
My bad; leftover. Done!